icefoganalytics/travel-authorization
# Add General Ledger (G/L) Coding Entry Widget to Travel Authorization Expense Tab

Fixes https://github.com/icefoganalytics/travel-authorization/issues/144

Relates to:
- [Analysis, Expense Pages, 2023-12-19](https://docs.google.com/document/d/1cWEA-0w9ro1TyajfnffkYaKL42yYnA89i9gdQysht_A/edit?pli=1#heading=h.zc2nwn7nmjlu)

# Context

**Is your feature request related to a problem? Please describe.**
The user needs an interface to enter General Ledger (G/L) coding information for travel authorizations expenses.

**Describe the solution you'd like**
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/a5850d24-b113-498b-9570-4936587f202a)

**Additional context**
- This will be simpler than the UI in the wireframe, just a string field and a float field.
- There will be no validation applied at this time.

# Implementation

Add `vue2-helpers` library so that I can use vue-router in <script setup> contexts.
Add "use" support to snack plugin so I can use it in <script setup> contexts.
Add a "use" file for general ledger codings instead of a Vuex Store because "use" files are more flexible, and I don't think the speed up gained by front-end cachine in Vuex stores offsets the additional complexity.

# Screenshots

![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/8dcb561d-1011-4b40-be3c-a95b37e8afaa)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "My Travel Requests" page from the top nav drop down.
5. Create a new travel authorization via the "+ Travel Authorization" button.
6. Fill in the various sections, and then click "Generate Estimate".
8. Submit the form to yourself.
9. Go to the Manager View from the top drop down nav.
10. Find your travel request in the top left widget.
11. Click on it and then scroll to the bottom and click Approve.
12. Go back to the "My Travel Requests" page from the top nav drop down.
13. Select the travel authorization you previously created.
14. Go to Expense tab and scroll to the bottom.
15. Check that you can add, update, and delete value(s) to the new Coding table.
